ether/core.py

```diff
@@ -50,33 +75,48 @@ class Node:
     capacity: Capacity
     arch: str
     labels: Dict[str, str]
+    coordinate: Optional[Coordinate]
 
     def __init__(self, name: str, capacity: Capacity = None, arch='x86', labels: Dict[str, str] = None) -> None:
         super().__init__()
         self.name = name
         self.capacity = capacity or Capacity()
         self.arch = arch
         self.labels = labels or dict()
+        self.coordinate = None
 
     def __repr__(self):
         return self.name
 
+    def distance_to(self, other: 'Node') -> float:
+        if self.coordinate is None:
+            raise AssertionError('node has no coordinate set')
+        if other.coordinate is None:
+            raise AssertionError('other node has no coordinate set')
+
+        return self.coordinate.distance_to(other.coordinate)
+
 
 class Route:
     source: Node
     destination: Node
+    path: list
```

Comment:
maybe keep the `List` type hint? TTBOMK using `list` as type hint is new in python 3.9

Reply:
I can't remember why, but seems to be a typo rather than a deliberate choice. I'm using `List[...]` everywhere else, so I applied your suggested change to be consistent.